File: WebApp/WebApp/Models/Reservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using WebApp.Dtos.Flight;

namespace WebApp.Models
{
    public class Reservation
    {
        public int Id { get; set; }
        [Required]
        public User User { get; set; }
        [Required]
        public Flight DepartingFlight { get; set; }
        public Flight ReturningFlight { get; set; }
    }
}

File: angular-app/src/models/Reservation.ts
import { Flight } from 'src/models/Flight';
import { User } from 'src/models/User';

export class Reservation {
    public constructor(init?: Partial<Reservation>) {
        Object.assign(this, init);
    }

    Id: number;
    User: User;
    DepartingFlight: Flight;
    ReturningFlight: Flight;
}